pkg/azuredisk/controllerserver.go

```diff
@@ -435,7 +436,61 @@ func (d *Driver) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest) (
 
 // ControllerExpandVolume controller expand volume
 func (d *Driver) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "ControllerExpandVolume is not yet implemented")
+	if len(req.GetVolumeId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID missing in request")
+	}
+	if err := d.ValidateControllerServiceRequest(csi.ControllerServiceCapability_RPC_EXPAND_VOLUME); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid expand volume request: %v", req)
+	}
+
+	capacityBytes := req.GetCapacityRange().GetRequiredBytes()
+	if capacityBytes == 0 {
+		return nil, status.Error(codes.InvalidArgument, "volume capacity range missing in request")
+	}
+	requestSize := *resource.NewQuantity(capacityBytes, resource.BinarySI)
+
+	diskURI := req.GetVolumeId()
+	if err := isValidDiskURI(diskURI); err != nil {
```

Comment:
no need to check here since `isManagedDisk` already did the check

Reply:
```go
func isValidDiskURI(diskURI string) error {
	if isManagedDisk(diskURI) {
		if strings.Index(diskURI, "/subscriptions/") != 0 {
			return fmt.Errorf("Inavlid DiskURI: %v, correct format: %v", diskURI, diskURISupportedManaged)
		}
	} else {
```
```go
func isManagedDisk(diskURI string) bool {
	if len(diskURI) > 4 && strings.ToLower(diskURI[:4]) == "http" {
		return false
	}
	return true
}
```
`isManagedDisk` does not check the format. So here we need `isValidDiskURI` to check it.
Then if the URI which is not a ManagedDisk URI can also pass `isValidDiskURI`, we should use `isManagedDisk` to check whether it is a ManagedDisk URI.

Reply:
then pls also add the check in `isManagedDisk` func, current `isManagedDisk` func is too simple, we should only need one function(`isManagedDisk`) to check the managed disk format

Reply:
I think we do not need to change the logic. The duties of the two functions are divided clear now. 
`isManagedDisk ` is just used to distinguish `managedDisk` from `Blob`. It should not return error info.  
`isValidDiskURI ` is responsible for checking the format of both kinds.

So here I just changed the order of checking. Now it seems to be more reasonable.
Firstly, judge the URI if it is a managed URI. Secondly, check its validation.